src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java

```diff
@@ -133,18 +134,20 @@ public synchronized void setUp(Context context, Run<?, ?> build, FilePath worksp
         // Set Environment variable
 
         setEnv(initialEnvironment);
-        String nodeSpecificFileSep = getNodeSpecificFileSeperator(launcher);
+
+        FilePath matlabExecutablePath = new FilePath(launcher.getChannel(),
+                getLocalMatlab() + "/bin/" + getNodeSpecificExecutable(launcher));
+
+        if (!matlabExecutablePath.exists()) {
+            throw new MatlabNotFoundError(Message.getValue("matlab.not.found.error"));
+        }
         // Add "matlabroot" without bin as env variable which will be available across the build.
         context.env("matlabroot", getLocalMatlab());
         // Add matlab bin to path to invoke MATLAB directly on command line.
-        context.env("PATH+matlabroot", getLocalMatlab() + nodeSpecificFileSep + "bin");     
+        context.env("PATH+matlabroot", getLocalMatlab() + "/bin"); 

```

Comment:
Would it be better to just matlabExecutablePath.parent() ? This way we are using the FilePath abstraction and reusing our previously computed matlabExecutablePath?

Reply:
Done.